service/src/MSRIOGroup.cpp

```diff
@@ -200,10 +179,76 @@ namespace geopm
         register_rdt_signals();
 
         register_control_alias("POWER_PACKAGE_LIMIT", "MSR::PKG_POWER_LIMIT:PL1_POWER_LIMIT");
-        register_control_alias("CPU_FREQUENCY_CONTROL", "MSR::PERF_CTL:FREQ");
         register_control_alias("POWER_PACKAGE_TIME_WINDOW", "MSR::PKG_POWER_LIMIT:PL1_TIME_WINDOW");
     }
 
+    void MSRIOGroup::register_frequency_signals(void)
+    {
+        // HWP vs P-State signals
+        if (m_hwp_is_enabled){
+            register_signal_alias("CPU_FREQUENCY_STATUS", "MSR::PERF_STATUS:FREQ");
+
+            register_signal_alias("CPU_FREQUENCY_CONTROL", "MSR::HWP_REQUEST:MAXIMUM_PERFORMANCE");
```

Comment:
Signal alias for CPU_FREQUENCY_CONTROL should match the control alias (i.e. both point to max, or both point to desired)